tar/ccache/ccache_read.c

```diff
@@ -65,7 +68,17 @@ read_rec(void * cookie)
 	ccr->ino = (ino_t)le64dec(ccre.ino);
 	ccr->size = (off_t)le64dec(ccre.size);
 	ccr->mtime = (time_t)le64dec(ccre.mtime);
+#if SIZE_MAX < UINT64_MAX
+	ccr_nch = le64dec(ccre.nch);
```

Comment:
Does that indicate a corrupt cache file?  In `ccache_entry.c` `callback_addchunk()`, I don't see any 32-bit-related maximum size for `->nch`.  So as far as I can tell, a cache file could be perfectly valid on a 64-bit platform, but not valid on a 32-bit platform.

If I adjust the existing sanity check, people might get a "Cache file is corrupt" message when actually it would still be valid on a 64-bit platform.  I thought it would be nice to have a separate error message for this case.